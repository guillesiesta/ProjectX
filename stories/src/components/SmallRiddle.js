import React, {Component} from 'react';

export default class SmallRiddle extends Component {

handleRiddle(t){
  this.props.onSubmit(t);
}
  render(){
    return(
      <form onSubmit={()=> this.handleRiddle(true)}>
        <div className="box-header with-border">
          <div className="row">
            <strong>Storie 1 SmallRiddle</strong>
            <p>LoremLorem Ipsum es simplemente el texto de relleno de
              las imprentas y archivos de texto. Lorem Ipsum ha sido el
              texto de relleno estándar de las industrias desde el año 1500,
            </p>
            <button type="submit" className="btn btn-default">Acceder</button>
          </div>
        </div>
      </form>
    )
  }
}
