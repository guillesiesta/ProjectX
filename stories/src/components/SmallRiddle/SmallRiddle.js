import React, {Component} from 'react';

export default class SmallRiddle extends Component {
  constructor(props){
    super(props);
    this.state={
      acertijo:'por defecto', //se almacena el acertijo
    };
  }
handleRiddle(t){ //devuelvo hacia atrás el título del acertijo clicado
  this.props.onSubmit(t);
}

componentDidMount(){
  //console.log("datos de atras:"+this.props.value)
  fetch('https://projectx-wvueafqhpp.now.sh/acertijo_por_titulo', {
        method: 'POST', // or 'PUT'
        body: JSON.stringify(this.props.value), // data can be `string` or {object}!
        headers:{
          'Content-Type': 'application/json'
        }})
    .then(response => response.json())
    .then(data => {
      //console.log(data[0].nick) // Prints result from `response.json()` in getRequest
      //console.log(data[0].nick)
      //console.log(data[0].password)
      //console.log(data[0].short_storie)
      this.setState({acertijo:data[0].short_storie})

    })
    .catch(error => console.error(error))

}
cargaAcertijo(){
  return(
    <p>{this.state.acertijo}</p>
  )
}
  render(){
    return(
      <form onSubmit={()=> this.handleRiddle(this.props.value)}>
        <div className="box-header with-border">
          <div className="row">
            <strong>{this.props.value}</strong>
            {this.cargaAcertijo()}
            <button type="submit" className="btn btn-default">Acceder</button>
          </div>
        </div>
      </form>
    )
  }
}
