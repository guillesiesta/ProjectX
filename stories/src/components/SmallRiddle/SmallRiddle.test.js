import React, {Component} from 'react';
import { shallow } from 'enzyme';

import SmallRiddle from './SmallRiddle';

import "isomorphic-fetch"

describe('SmallRiddle', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <SmallRiddle />
    );

    expect(wrapper).toMatchSnapshot();
  })
})
