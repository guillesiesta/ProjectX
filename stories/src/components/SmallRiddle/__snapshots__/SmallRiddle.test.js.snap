// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallRiddle Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SmallRiddle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-header with-border"
      >
        <div
          className="row"
        >
          <strong />
          <p>
            por defecto
          </p>
          <button
            className="btn btn-default"
            type="submit"
          >
            Acceder
          </button>
        </div>
      </div>,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <strong />
          <p>
            por defecto
          </p>
          <button
            className="btn btn-default"
            type="submit"
          >
            Acceder
          </button>
        </div>,
        "className": "box-header with-border",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong />,
            <p>
              por defecto
            </p>,
            <button
              className="btn btn-default"
              type="submit"
            >
              Acceder
            </button>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "por defecto",
            },
            "ref": null,
            "rendered": "por defecto",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Acceder",
              "className": "btn btn-default",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Acceder",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-header with-border"
        >
          <div
            className="row"
          >
            <strong />
            <p>
              por defecto
            </p>
            <button
              className="btn btn-default"
              type="submit"
            >
              Acceder
            </button>
          </div>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <strong />
            <p>
              por defecto
            </p>
            <button
              className="btn btn-default"
              type="submit"
            >
              Acceder
            </button>
          </div>,
          "className": "box-header with-border",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong />,
              <p>
                por defecto
              </p>,
              <button
                className="btn btn-default"
                type="submit"
              >
                Acceder
              </button>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "por defecto",
              },
              "ref": null,
              "rendered": "por defecto",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Acceder",
                "className": "btn btn-default",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Acceder",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
